lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -394,6 +394,104 @@
 
 INSTANCE_ATTRIBUTES = ('instance_id', 'health_status', 'lifecycle_state', 'launch_config_name')
 
+backoff_params = dict(tries=10, delay=3, backoff=1.5)
+
+@AWSRetry.backoff(**backoff_params)
+def describe_autoscaling_groups(connection, group_name):
+    response = connection.describe_auto_scaling_groups(AutoScalingGroupNames=[group_name])
```

Comment:
This should be paginated: http://boto3.readthedocs.io/en/latest/reference/services/autoscaling.html#paginators